### PR TITLE
End device level payload formatters

### LINKED ISSUE #778: End Device level payload formatters
#### Summary
Complementary to application level payload formatters (introduced via #742), the console needs functionality to manage payload formatters on end device level, which, if set, take precedence.

#### Why do we need this?
Basic functionality

#### What is already there? What do you see now?
Application level payload formatters

#### What is missing? What do you want to see?
End device level payload formatters

#### Environment
Console in browser

#### How do you propose to implement this?
Complementary to the implementation for applications

#### Can you do this yourself and submit a Pull Request?
Yes.

### PR DESCRIPTION
**Pull request by kschiffer**
#### Summary
Closes #778 

#### Changes
- Modify `<PayloadFormatterForm />` to reduce repetition
- Introduce `<DevicePayloadFormatter />`-container
- Add device payload formatter view and route
- Add/modify redux logic for updating devices
- Fix path comparison bug in device update service
- Introduce request redux middleware to return promises from action creators
- Introduce `createRequestLogic` utility function to generate decorated request logics
- Various minor related styling fixes

#### Notes for Reviewers
This PR contains the new redux request action workflow which we discussed. I'm wondering whether I should source this out to another PR rater. Made this a draft until we found a way to go here.
See also #800 

#### Release Notes
- Added end device level payload formatters to console

### CODE REVIEW

File: pkg/webui/console/views/device/index.js [hunk 1]
```diff
@@ -137,18 +150,20 @@ export default class Device extends React.Component {
...
               <Tabs
+                className={style.tabs}
                 narrow
                 tabs={tabs}
-                className={style.tabs}
               />
             </Col>
           </Row>
         </Container>
+        <hr className={style.rule} />
```
[the remark this answers is not in the document]
Comment (kschiffer) [reply pkg/webui/console/views/device/index.js]: It's a special styling to make the horizontal rule below the tabs in the device subview span the whole width of the container. Makes for a better distinction of the different sections in the payload formatter view. It's also complying to the screen design this way.

[the remark this answers is not in the document]
Comment (bafonins) [reply pkg/webui/console/views/device/index.js]: Screenshot 2019-06-07 at 10 17 59
is this the reason for the tabs taking `100%` of the width?

[the remark this answers is not in the document]
Comment (kschiffer) [reply pkg/webui/console/views/device/index.js]: Yes.

File: pkg/webui/console/views/device/index.js [hunk 2]
```diff
@@ -119,11 +127,16 @@ export default class Device extends React.Component {
 
     const basePath = `/console/applications/${appId}/devices/${devId}`
 
+    // Prevent default redirect to uplink when tab is already open
+    const payloadFormattersLink =
+    pathname.startsWith(`${basePath}/payload-formatters`)
```
[the remark this answers is not in the document]
Comment (kschiffer) [reply pkg/webui/console/views/device/index.js]: But it does not result in any problems here, or what do I miss?

**Pull request merged**